streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -322,14 +323,16 @@ void setConsumer(final Consumer<byte[], byte[]> consumer) {
      * @throws StreamsException if the store's change log does not contain the partition
      */
     boolean updateNewAndRestoringTasks() {
+        // wait for other thread in the same jvm to free the locks of state store
+        Utils.sleep(10);
         active.initializeNewTasks();
         standby.initializeNewTasks();
 
         final Collection<TopicPartition> restored = changelogReader.restore(active);
 
         active.updateRestored(restored);
 
-        if (active.allTasksRunning()) {
+        if (active.allTasksRunning() && standby.allTasksRunning()) {
```

Comment:
I think the logic here is not correct: we should still resume the main consumer and assign standby partitions if active tasks are all running; we should only alter the logic of returning flag with both active / standby all running.

Reply:
Thanks @guozhangwang,  I will fix the logic in next commit.